### PR TITLE
feat(api): license service shell

### PR DESCRIPTION
**Pull request by osk**
# License service "shell"

https://www.notion.so/License-service-WIP-d8a9df75b7d747d1950b7c155507ab7c

## What

Api endpoint to deliver all licenses a user has (Ökuleyfi, Veiðileyfi, Vinnuvélaskírteini, Skotvopnaleyfi o.fl.)

This initial release only includes drivers license

## Screenshots / Gifs

https://www.figma.com/file/kzvPtwIM25MrBmoKOhEkji/S%C3%8D-app?node-id=963%3A0

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by darrikonn — DISMISSED** [review]
There's already a domain for this (with a better description, license is too broad): `libs/api/domains/driving-license` ... it already contains an almost complete implementation of this.

**Comment by osk** [comment]
@darrikonn thanks, we're still feeling out how this shell for different license services will work. It'll contain many more eventually and need a generic way to handle them.
See more here: https://www.notion.so/License-service-WIP-d8a9df75b7d747d1950b7c155507ab7c
Sorry, didn't intend to get eyes on this right away.

**Comment by darrikonn** [comment]
> @darrikonn thanks, we're still feeling out how this shell for different license services will work. It'll contain many more eventually and need a generic way to handle them.
> See more here: [notion.so/License-service-WIP-d8a9df75b7d747d1950b7c155507ab7c](https://www.notion.so/License-service-WIP-d8a9df75b7d747d1950b7c155507ab7c)
> Sorry, didn't intend to get eyes on this right away.

Ahh ok nice!

**Review by Valur — APPROVED** [review]
Initial release for license cards in app

File: libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseMappers.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+          },
+          {
+            type: GenericLicenseDataFieldType.Value,
+            label: 'Útgáfudagur',
+            value: field.utgafuDags,
+          },
+        ],
+      })),
+    } as GenericLicenseDataField,
+  ].filter((Boolean as unknown) as ExcludesFalse)
```
Comment (eirikurn) [inline libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseMappers.ts:128]: Is this a logic error? Doesn't look like this array will ever contain false values.

File: libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseMappers.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+  // We parse license data into the fields as they're displayed on the physical drivers license
+  // see: https://www.samgongustofa.is/umferd/nam-og-rettindi/skirteini-og-rettindi/okurettindi-og-skirteini/
+  const data: Array<GenericLicenseDataField> = [
+    // We don't get the name split into two from the API, combine
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: '2. Eiginnafn 1. Kenninafn',
+      value: license.nafn,
+    } as GenericLicenseDataField,
```
Comment (eirikurn) [inline libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseMappers.ts:73]: Do you need this type cast if the overall variable has a type definition?

File: libs/api/domains/license-service/src/lib/licenceService.type.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  // TODO(osk) should these be here? or be resolved by client via contentful?
+  // Commented out until talked about, to limit scope of v1
+  /*
+  title: string
+  ordering: number
+  backgroundImage?: string
+  applicationUrl?: string
+  detailUrl?: string
+  */
```
Comment (eirikurn) [inline libs/api/domains/license-service/src/lib/licenceService.type.ts:69]: Let's make sure we get rid of all commented code asap.

**Review by eirikurn — APPROVED** [review]

File: libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseMappers.ts (hunk 1 above)
Comment (osk) [reply libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseMappers.ts:128]: yes, this and the casts are redundant as is

File: libs/api/domains/license-service/src/lib/licenceService.type.ts (hunk 3 above)
Comment (osk) [reply libs/api/domains/license-service/src/lib/licenceService.type.ts:69]: yes! starting a new branch right away

File: libs/api/domains/license-service/src/lib/licenseService.service.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    if (!drivingLicense) {
+      // TODO(osk) how do we handle null?
+      throw Error(`unable to get drivers license for nationalId ${nationalId}`)
+    }
+
+    const genericDrivingLicense = drivingLicensesToSingleGenericLicense(
+      drivingLicense,
+    )
+    console.log({ drivingLicense })
```
Comment (atligudl) [inline libs/api/domains/license-service/src/lib/licenseService.service.ts:68]: Console log lost in here

File: libs/service-portal/core/src/lib/navigation/masterNavigation.ts [hunk 5]
```diff
@@ -176,6 +176,19 @@ export const servicePortalMasterNavigation: ServicePortalNavigationItem[] = [
         },
       },
 
+      {
+        name: defineMessage({
+          id: 'service.portal:licenses',
+          defaultMessage: 'Skilríki',
+        }),
+        path: ServicePortalPath.LicensesRoot,
+        systemRoute: true,
```
Comment (atligudl) [inline libs/service-portal/core/src/lib/navigation/masterNavigation.ts:185]: Setting this flag to true means to this item will always be included in the navigation whether a route exists for it or not.
This property should probably have been made more descriptive in hindsight.

See https://github.com/island-is/island.is/blob/main/libs/service-portal/core/src/lib/service-portal-core.ts

**Review by atligudl — APPROVED** [review]

File: libs/api/domains/license-service/src/lib/licenseService.service.ts (hunk 4 above)
Comment (Valur) [reply libs/api/domains/license-service/src/lib/licenseService.service.ts]: Removed

File: libs/service-portal/core/src/lib/navigation/masterNavigation.ts (hunk 5 above)
Comment (Valur) [reply libs/service-portal/core/src/lib/navigation/masterNavigation.ts]: Fixed

**Pull request merged**